### ISSUE TITLE
Option to ignore failed plugin?

### ISSUE THREAD

**Issue by iisworks** [coredns/coredns#4600]
We are testing Coredns on Ubuntu 18.04  in Azure, using the Azure plugin to load 50 Azure Private DNS zones. I noticed that when one of these zone resources gets removed from Azure while it is still being referenced in the Corefile, Coredns does not start up any more at all. Is there any way of preventing an individual failed zone plugin/zone from blocking the entire server to limit the impact of a misconfiguration? Is it possible to fail/ignore only that specific zone somehow?

We are using the Azure plugin to serve Privatelink zones using Coredns, with "Fallthrough" to Azure Dns for records that are not found in the zone.   Info: https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-dns

**Comment by miekg** [comment]
[ Quoting ***@***.***> in "[coredns/coredns] Option to ignore ..." ]
>We are testing Coredns on Ubuntu 18.04 in Azure, using the Azure plugin to load
>50 Azure Private DNS zones. I noticed that when one of these zone resources
>gets removed from Azure while it is still being referenced in the Corefile,
>Coredns does not start up any more at all. Is there any way of preventing an
>individual failed zone plugin/zone from blocking the entire server to limit the
>impact of a misconfiguration? Is it possible to fail/ignore only that specific
>zone somehow?
>
>We are using the Azure plugin to serve Privatelink zones using Coredns, with
>"Fallthrough" to Azure Dns for records that are not found in the zone. Info:
>https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-dns

that's probably a bug in that plugin. Most (all?) plugins should try to get their state
even after startup and not block the entire server from starting.

ignoring plugins is not implemented

**Comment by iisworks** [comment]
Thanks, I will double-check this issue and report back! 
BTW: I may have only checked the Health probe (8080), does that change when one plugin fails to load?

[Update] Just performed a quick test while logged in on the VM, and indeed the Coredns service does not start after a Private DNS Zone resource was deleted in Azure. When I removed the entry from Corefile, the Coredns service started again. I created a bug to get this checked out.
